lightning/src/ln/channelmanager.rs

```diff
@@ -8931,6 +9013,140 @@ where
+		// TODO(splicing): checks
+		// TODO(splicing): if we accept splicing, quiscence
+
+		let per_peer_state = self.per_peer_state.read().unwrap();
+		let peer_state_mutex = per_peer_state.get(counterparty_node_id)
+			.ok_or_else(|| {
+				debug_assert!(false);
+				MsgHandleErrInternal::send_err_msg_no_close(format!("Can't find a peer matching the passed counterparty node_id {}", counterparty_node_id), msg.channel_id)
+			})?;
+		let mut peer_state_lock = peer_state_mutex.lock().unwrap();
+		let peer_state = &mut *peer_state_lock;
+
+		let our_funding_contribution = 0i64;
+
+		// Look for the channel
+		match peer_state.channel_by_id.entry(msg.channel_id) {
+			hash_map::Entry::Vacant(_) => return Err(MsgHandleErrInternal::send_err_msg_no_close(format!(
+					"Got a message for a channel from the wrong node! No such channel for the passed counterparty_node_id {}, channel_id {}",
+					counterparty_node_id, msg.channel_id,
+				), msg.channel_id)),
+			hash_map::Entry::Occupied(mut chan_entry) => {
+				if let ChannelPhase::Funded(chan) = chan_entry.get_mut() {
+					let pre_channel_value = chan.context.get_value_satoshis();
+					// Sanity check: capacity cannot decrease below 0
+					if (pre_channel_value as i64).saturating_add(msg.funding_contribution_satoshis) < 0 {
+						return Err(MsgHandleErrInternal::send_err_msg_no_close(format!(
+							"Post-splicing channel value cannot be negative. It was {} - -{}", pre_channel_value, msg.funding_contribution_satoshis,
+						), msg.channel_id));
+					}
+
+					if msg.funding_contribution_satoshis < 0 {
+						return Err(MsgHandleErrInternal::send_err_msg_no_close(format!(
+							"Splice-out not supported, only splice in, relative {}", -msg.funding_contribution_satoshis,
+						), msg.channel_id));
+					}
+
+					let post_channel_value = PendingSpliceInfoPre::compute_post_value(pre_channel_value, msg.funding_contribution_satoshis, our_funding_contribution);
+
+					// Early check for reserve requirement, assuming maximum balance of full channel value
+					// This will also be checked later at tx_complete
+					let _res = chan.context.check_balance_meets_reserve_requirements(post_channel_value, post_channel_value)
+						.map_err(|e| MsgHandleErrInternal::from_chan_no_close(e, msg.channel_id))?;
+
+					// Check if a splice has been initiated already.
+					// Note: this could be handled more nicely, and support multiple outstanding splice's, the incoming splice_ack matters anyways.
+					if let Some(splice_info) = &chan.context.pending_splice_pre {
+						return Err(MsgHandleErrInternal::send_err_msg_no_close(format!(
+							"Channel has already a splice pending, channel id {}, contribution {}",
+							msg.channel_id, splice_info.our_funding_contribution,
+						), msg.channel_id));
+					}
+
+					// TODO(splicing): Below step should come after channel phase change (after re-adding to channel map)
```

Comment:
Do we ever remove it from the channel map? Was thinking we were just keeping it in the funded phase.

Reply:
Not as of now, removed TODO comment